ml-agents/mlagents/trainers/trainer.py

```diff
@@ -20,22 +21,33 @@ class UnityTrainerException(UnityException):
 class Trainer(object):
     """This class is the base class for the mlagents.trainers"""
 
-    def __init__(self, brain, trainer_parameters, training, run_id):
+    def __init__(self, brain, trainer_parameters, training, run_id,
+                 debug_flag):
         """
         Responsible for collecting experiences and training a neural network model.
         :BrainParameters brain: Brain to be trained.
         :dict trainer_parameters: The parameters for the trainer (dictionary).
         :bool training: Whether the trainer is set for training.
         :int run_id: The identifier of the current run
+        :bool debug_flag: Checks debug_flag
         """
         self.param_keys = []
         self.brain_name = brain.brain_name
         self.run_id = run_id
         self.trainer_parameters = trainer_parameters
+        self.time_start_experience_collection = None
```

Comment:
If these added fields are purely for logging rather than added trainer functionality it would probably make sense to keep them in a separate code path (e.g. a separate class)

Reply:
PTAL